python/GafferUI/EditScopeUI.py

```diff
@@ -177,19 +200,51 @@ def _updateFromValues( self, values, exception ) :
 			self._qtWidget().setProperty( "editScopeActive", GafferUI._Variant.toVariant( editScopeActive ) )
 			self._repolish()
 
-	def __updateMenuButton( self, editScope ) :
+	def __updatePlugInputChangedConnection( self ) :
+
+		self.__plugInputChangedConnection = self.getPlug().node().plugInputChangedSignal().connect(
+			Gaffer.WeakMethod( self.__plugInputChanged ), scoped = True
+		)
+
+	def __plugInputChanged( self, plug ) :
+
+		if plug.getName() == "in" and plug.parent() == self.getPlug().node() :
+			# The result of `__inputNode()` will have changed.
+			self.__acquireContextTracker()
+			self.__updateMenuButton()
```

Comment:
We can see a bit of a flicker in the menu icon when the input changes between two plugs downstream of the current Edit Scope. When the input changes, the icon can quickly toggle between "warningSmall.png" and the swatch.

I assume because the context tracker isn't guaranteed to be up to date at the time `__updateMenuButton()` is called here, resulting in "warningSmall.png" being used as the Edit Scope isn't currently tracked, then shortly afterwards `__updateMenuButton()` is called again from `__contextTrackerChanged()`, now with the Edit Scope tracked and we change back to displaying the swatch.

Can we avoid calling `__updateMenuButton()` here and rely on `__contextTrackerChanged()` updating the icon once we have a more accurate view of the upstream nodes?

Reply:
Addressed in https://github.com/GafferHQ/gaffer/pull/5967/commits/24accc30ce1b146b6447a455170188f35bcb3334. If that looks OK, it'll just need squashing in before we merge.